product_sticker/models/product_product.py

```diff
@@ -0,0 +1,24 @@
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _get_sticker_arguments(self):
+        pavs = self.product_template_variant_value_ids.product_attribute_value_id
+        return {
+            "categories": self.categ_id,
+            "attributes": pavs.attribute_id,
+            "attribute_values": pavs,
+        }
+
+    def get_product_stickers(self):
+        """Product Stickers related to this Product Variant and his Template"""
+        self.ensure_one()
+        # Product Template: Common stickers
+        pt_stickers = self.product_tmpl_id.get_product_stickers()
+        # Product Product: Specific stickers
+        pp_stickers = self.env["product.sticker"].get_stickers(
+            **self._get_sticker_arguments()
+        )
+        return pt_stickers | pp_stickers
```

Comment:
We shouldn't return recordsets in public methods. These methods can be called via RPC, and a recordset doesn't make sense in that context.

Please either make the method private or return ids. Or you could add the `@api.returns` decorator on them, to let Odoo know what to do.